llvm/lib/Transforms/IPO/MergeFunctions.cpp

```diff
@@ -297,17 +300,31 @@ class MergeFunctions {
   // dangling iterators into FnTree. The invariant that preserves this is that
   // there is exactly one mapping F -> FN for each FunctionNode FN in FnTree.
   DenseMap<AssertingVH<Function>, FnTreeType::iterator> FNodesInTree;
+
+  /// Deleted-New functions mapping
+  std::map<Function *, Function *> DelToNewMap;
```

Comment:
This (and probably other maps here) could be a `DenseMap` instead, unless there's a reason this needs `std::map`. I think it could also be a returned value from `runOnFunctions` rather than a member, but that's mostly subjective.

Reply:
Resolved in https://github.com/llvm/llvm-project/pull/111045/commits/67d70af9d4e0f1d09f77d8eae77ea5c420bc91df .